Chained.h

```diff
@@ -11,197 +13,195 @@
-        for(auto kvp : (*hashmap)[index]){
-            if(k == kvp.key){
-                return kvp.value;
+    Value get(Key k) override {
+        uint64_t index = hash(k, hashmap_.size());
+        for(auto kvp : hashmap_[index]){
+            if(k == kvp.key_){
+                return kvp.value_;
             }
         }
-        return defaultGetReturn();
+        return Value{};
     }
 
-    void put(Key k, Value v) {
-        uint64_t index = hash(k);
-        for(auto kvp : (*hashmap)[index]){
-            if(kvp.key == k){
-                kvp.value = v;
+    void put(Key k, Value v) override {
+        uint64_t index = hash(k, hashmap_.size());
+        for(auto kvp : hashmap_[index]){
+            if(kvp.key_ == k){
+                kvp.value_ = v;
                 return;
             }
         }
         KeyValuePair kvp(k, v);
-        (*hashmap)[index].push_back(kvp);
-        ++numPairs;
-        if((static_cast<float>(numPairs))/(static_cast<float>(hashmapSize)) > MAX_LOAD_FACTOR){
+        hashmap_[index].push_back(kvp);
+        ++numPairs_;
+        if(getLoadFactor() > MAX_LOAD_FACTOR_){
             doubleSize();
         }
     }
 
-    void remove(Key k) {
-        uint64_t index = hash(k);
-        for(auto i = (*hashmap)[index].begin(); i != (*hashmap)[index].end(); ++i) {
-            if(k == i->key){
-                (*hashmap)[index].erase(i);
-                --numPairs;
+    void remove(Key k) override {
+        uint64_t index = hash(k, hashmap_.size());
+        for(auto i = hashmap_[index].begin(); i != hashmap_[index].end(); ++i) {
+            if(k == i->key_){
+                hashmap_[index].erase(i);
+                --numPairs_;
                 return;
             }
         }
+        std::cerr << "Chained Map does not contain the key \"" << k << "\" so it could not be removed\n";
     }
 
-    void display(bool print){
+    void display() override {
+        for(uint64_t i = 0; i < hashmap_.size(); i++){
+            std::cout << i << ": |" << hashmap_[i].size() << "| ";
+            for(auto kvp : hashmap_[i]){
+                    printKeyValuePair(kvp);
+            }
+                std::cout << "\n";
+        }
+        std::cout << std::endl;
+    }
+
+    void stats() override {
         std::vector<uint64_t> elements;
         std::vector<uint64_t> collisions;
         uint64_t numWith0 = 0;
         uint64_t numWith1 = 0;
         uint64_t numWithMore = 0;
 
-        //loop through and print the data structure while storing important information
-        for(uint64_t i = 0; i < hashmapSize; i++){
-            if(print) {
-                std::cout << i << ": |" << (*hashmap)[i].size() << "| ";
-            }
-            elements.push_back(static_cast<uint64_t>((*hashmap)[i].size()));
-            if((*hashmap)[i].empty()) {
+        for(uint64_t i = 0; i < hashmap_.size(); i++){
+            elements.push_back(static_cast<uint64_t>(hashmap_[i].size()));
```

Comment:
Yeah, it is good to cast a `size_t `to a `unit64_t`.  I had to think about that one.